### PR TITLE
New proxy setup

### PR DESCRIPTION
**Pull request by Lezek123**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -58,15 +81,15 @@ services:
...
+      # - /var/run/docker.sock:/var/run/docker.sock
       - ./local/logs:/youtube-synch/local/logs
       - ./local/data:/youtube-synch/local/data
       # mount Google Cloud's Application Default Credentials file. A default bind mount is created
       # as workaround for scenario when `YT_SYNCH__YOUTUBE__ADC_KEY_FILE_PATH` will be undefined,
       # since docker-compose does not support creating bind-mount volume with empty path.
       - ${YT_SYNCH__YOUTUBE__ADC_KEY_FILE_PATH:-./package.json}:${YT_SYNCH__YOUTUBE__ADC_KEY_FILE_PATH:-/package.json}:ro
       # mount the AWS credentials file to docker container home directory
-      - ~/.aws/:/root/.aws:ro
+      # - ~/.aws/:/root/.aws:ro
```
Comment (mnaamani) [inline docker-compose.yml:92]: The AWS credentials are loaded from .env? That is why we don't need to mount the aws folder?

File: src/services/youtube/api.ts [hunk 2]
```diff
@@ -396,20 +397,34 @@ class YoutubeClient implements IYoutubeApi {
...
+    console.error(`Using proxy ${proxy} to download ${videoUrl}...`)
     const response = await ytdl(videoUrl, {
       noWarnings: true,
       printJson: true,
       format: 'bv*[height<=1080][ext=mp4]+ba[ext=m4a]/bv*[height<=1080][ext=webm]+ba[ext=webm]/best[height<=1080]',
       output: `${outPath}/%(id)s.%(ext)s`,
       ffmpegLocation: ffmpegInstaller.path,
       // forceIpv6: true,
-      limitRate: '4M',
+      limitRate: this.config.sync.limits?.bandwidthPerDownload,
```
Comment (mnaamani) [inline src/services/youtube/api.ts:424]: In my testing I didn't find the `limitRate` effectively limited the rate to desired value. It is the main metric by which we get blocked. And it blocks entire network range, not a single IP so we should to be sure.

File: docker-compose.yml (hunk 1 above)
Comment (Lezek123) [reply docker-compose.yml:70]: Yes

File: src/services/youtube/api.ts (hunk 2 above)
Comment (Lezek123) [reply src/services/youtube/api.ts]: Thank you for pointing that out, I changed the approach to use `proxychains` + `tc` for rate limiting instead.

**Pull request merged**